atc/creds/vault/manager.go

```diff
@@ -90,50 +86,27 @@ func (manager *VaultManager) MarshalJSON() ([]byte, error) {
 	})
 }
 
-func toString(s interface{}) string {
-	if s == nil {
-		return ""
-	}
-	return s.(string)
-}
+func (manager *VaultManager) Config(config map[string]interface{}) error {
+	// apply defaults
+	manager.PathPrefix = "/concourse"
+	manager.Auth.RetryMax = 5 * time.Minute
+	manager.Auth.RetryInitial = time.Second
```

Comment:
It's too bad that these are duplicating the defaults as declared in the struct tags above - is there some way, perhaps through [reflection magic](https://golang.org/pkg/reflect/#Value.FieldByName), to determine the defaults dynamically? Is it worth the readability cost?

Reply:
I thought about I think it's actually more likely that we'll end up with different defaults in the long run. It doesn't make much sense to me to have a default path prefix now that we're configuring Vault per-pipeline. So I don't think the maintenance overhead of reflection code is worth the slight duplication, since it'll be temporary in two ways: the defaults may intentionally become different between the two, and we'd probably sooner go exclusively `var_sources` than have them accidentally fall out of sink.

Reply:
i thought about this more today, and you also already have the 'double-entry' effect of the test for the defaults, so why unnecessarily complicate the implementation...